fala/apps/adviser/tests/test_search_view_function.py

```diff
@@ -17,6 +17,42 @@ def test_invalid_postcode_generates_error(self):
         self.assertEqual({"postcode": ["Postcode not found"]}, response.context_data["form"].errors)
 
 
+@override_settings(FEATURE_FLAG_NO_MAP=True)
+class PostCodeResultsPageTest(SimpleTestCase):
+    client = Client()
+    url = reverse("search")
+
+    data = {"postcode": "PE30", "name": "bu", "categories": "deb"}
+
+    def test_category_search_has_good_result(self):
+        response = self.client.get(self.url, self.data)
+        self.assertEqual(200, response.status_code)
+
+    def test_category_search_has_just_user_input(self):
+        response = self.client.get(self.url, self.data)
+        self.assertNotContains(response, "PE30 1AA")
+
+    def test_category_search_heading_closeness(self):
```

Comment:
It is not clear to me from these tests what they're intending to achieve. Are they just tests that particular strings are in the template, or is there logic that controls those strings? If the latter, my instinct is we should have counter-tests to demonstrate when they should be absent, otherwise I'm unsure of their value

Reply:
added some extra contra-tests

Reply:
They've not really done a lot for readability, have they?

Reply:
managed to come up with a slightly better solution